pkg/operator/vttablet/pod.go

```diff
@@ -370,6 +370,16 @@ func UpdatePod(obj *corev1.Pod, spec *Spec) {
 	}
 }
 
+func detectMySQLVersionUpgradeDowngrade() *corev1.Lifecycle {
```

Comment:
Shouldn't the name be `LifeCycleForMySQLVersionChange` or something along those line? What does the function `detect`?